test/common/io/io_uring_impl_test.cc

```diff
@@ -12,20 +12,36 @@ namespace {
 
 class IoUringBaseTest : public ::testing::Test {
 public:
-  IoUringBaseTest() : api_(Api::createApiForTest()), factory_(2, false, context_.threadLocal()) {
-    factory_.onServerInitialized();
+  IoUringBaseTest() : api_(Api::createApiForTest()) {
+    if (isIoUringSupported()) {
+      factory_ = std::make_unique<IoUringFactoryImpl>(2, false, context_.threadLocal());
+      factory_->onServerInitialized();
+    } else {
+      should_skip_ = true;
+    }
+  }
+
+  void SetUp() override {
+    if (should_skip_) {
+      GTEST_SKIP();
```

Comment:
Thinking out loud here, is there a way to ensure that CI does not skip these tests? (I realize that CI images support io_uring, but lets say some accidental change removes that support).
This will ensure that the code is checked at least in CI.

Reply:
Good point! I've added `source/common/io` to `test/per_file_coverage.sh`. So, skipped tests should be detected in the coverage task.